@cottsay flake8 is hitting this PR with an error that wasn't present in your PR:

```
  ----------------------------- Captured stdout call -----------------------------
  
  1     F824 `global spell_check_words_path` is unused: name is never assigned in scope
  ----------------------------- Captured stderr call -----------------------------
  /home/runner/work/colcon-cargo/colcon-cargo/test/test_spell_check.py:13:5: F824 `global spell_check_words_path` is unused: name is never assigned in scope
      global spell_check_words_path
      ^
  flake8 reported 1 errors
```

This isn't related to anything that I changed, so I'm guessing it's due to a flake8 update. I'm not sure if you'd like me to suppress the error or delete the spell check module or something else.

--- a/colcon_cargo/package_augmentation/cargo.py
+++ b/colcon_cargo/package_augmentation/cargo.py
@@ -90,7 +90,7 @@ def extract_dependencies(package_name, content, path):
         create_dependency_descriptor(k, v, path)
         for k, v in filter_dependency_list(
             content.get('dev-dependencies', {}).items(),
-            package_name,
+            filter_out=package_name,
         )
     }
     return {
@@ -99,7 +99,7 @@ def extract_dependencies(package_name, content, path):
     }
 
 
-def filter_dependency_list(dependencies, filter_out = None):
+def filter_dependency_list(dependencies, filter_out=None):
     filtered_dependencies = {}
     for dependency, constraints in dependencies:
         if isinstance(constraints, dict):

--- a/colcon_cargo/package_augmentation/cargo.py
+++ b/colcon_cargo/package_augmentation/cargo.py
@@ -73,22 +73,25 @@ def extract_dependencies(package_name, content, path):
     :returns: The dependencies
     :rtype: dict(string, set(DependencyDescriptor))
     """
-    target_name = content.get('name')
 
     depends = {
-        create_dependency_descriptor(target_name, k, v, path)
-        for k, v in content.get('dependencies', {}).items()
-        if k != target_name and k != package_name
+        create_dependency_descriptor(k, v, path)
+        for k, v in filter_dependency_list(
+            content.get('dependencies', {}).items()
+        )
     }
     build_depends = {
-        create_dependency_descriptor(target_name, k, v, path)
-        for k, v in content.get('build-dependencies', {}).items()
-        if k != target_name and k != package_name
+        create_dependency_descriptor(k, v, path)
+        for k, v in filter_dependency_list(
+            content.get('build-dependencies', {}).items()
+        )
     }
     dev_depends = {
-        create_dependency_descriptor(target_name, k, v, path)
-        for k, v in content.get('dev-dependencies', {}).items()
-        if k != target_name and k != package_name
+        create_dependency_descriptor(k, v, path)
+        for k, v in filter_dependency_list(
+            content.get('dev-dependencies', {}).items(),
+            package_name,
+        )
     }
     return {
         'build': depends | build_depends | dev_depends,
@@ -96,7 +99,19 @@ def extract_dependencies(package_name, content, path):
     }
 
 
-def create_dependency_descriptor(package, name, constraints, path):
+def filter_dependency_list(dependencies, filter_out = None):
+    filtered_dependencies = {}
+    for dependency, constraints in dependencies:
+        if isinstance(constraints, dict):
+            dependency = constraints.get('package', dependency)
+
+        if dependency != filter_out:
+            filtered_dependencies[dependency] = constraints
+
+    return filtered_dependencies.items()
+
+
+def create_dependency_descriptor(dependency_name, constraints, path):
     """
     Create a dependency descriptor from a Cargo dependency specification.
 
@@ -115,7 +130,6 @@ def create_dependency_descriptor(package, name, constraints, path):
         else:
             source = constraints.get('git') or \
                 constraints.get('registry')
-        name = constraints.get('package', name)
     else:
         source = None
     metadata = {
@@ -124,4 +138,4 @@ def create_dependency_descriptor(package, name, constraints, path):
     }
     # TODO: Interpret SemVer constraints and add appropriate constraint
     #       metadata. Handling arbitrary wildcards will be non-trivial.
-    return DependencyDescriptor(name, metadata=metadata)
+    return DependencyDescriptor(dependency_name, metadata=metadata)

--- a/test/test_spell_check.py
+++ b/test/test_spell_check.py
@@ -10,7 +10,6 @@ spell_check_words_path = Path(__file__).parent / 'spell_check.words'
 
 @pytest.fixture(scope='module')
 def known_words():
-    global spell_check_words_path
     return spell_check_words_path.read_text().splitlines()
 
 

